web/terraform-checkpoints/create-firebase-project-checkpoint.tf

```diff
@@ -0,0 +1,70 @@
+# Terraform configuration to set up providers by version.
+terraform {
+  required_providers {
+    google-beta = {
+      source  = "hashicorp/google-beta"
+      version = "~> 4.0"
+    }
+  }
+}
+
+# Configure the provider not to use the specified project for quota check.
+# This provider should only be used during project creation and initializing services.
+provider "google-beta" {
+  alias                 = "no_user_project_override"
+  user_project_override = false
+}
+
+# Configure the provider that uses the new project's quota.
+provider "google-beta" {
+  user_project_override = true
+}
+
+# Create a new Google Cloud project.
+resource "google_project" "default" {
+  provider = google-beta.no_user_project_override
+
+  # TODO: replace with your own values
+  name       = "<PROJECT_NAME_OF_YOUR_PROJECT>"
+  project_id = "<PROJECT_ID_OF_YOUR_PROJECT>"
+
+  # Required for the project to display in any list of Firebase projects.
+  labels = {
+    "firebase" = "enabled"
+  }
+}
+
+# Enable the required underlying Service Usage API.
```

Comment:
Do we really want the code-lab checkpoints to have the 'serviceusage' blocks to be unrolled?  it's a lot more compact to use the loops & just add new rows.

Reply:
I think we had to flatten it this way so it doesn't conflict with the for_each used in replicating the config across various envs (main-foreach.tf) below.

Reply:
Yes, if we use for_each on services, we end up needing a double for-loop for the combo of (prod, staging) and (serviceusage, firebase) for the project replication step.